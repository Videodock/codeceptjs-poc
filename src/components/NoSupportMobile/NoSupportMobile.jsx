import React from 'react';
import { makeStyles } from '@material-ui/styles';
import { useTranslation } from 'react-i18next';
import { Typography } from '@material-ui/core';
import images from '../../theme/images';
import styles from './NoSupportMobile.styles';

const useStyles = makeStyles(styles);

export default function NoSupportMobile () {
  const { t }   = useTranslation('noSupportMobile');
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <div className={classes.container}>
        <div className={classes.logoBlock}>
          <img className={classes.logoIcon} src={images.logoIcon} alt={t('logo_text')} />
          <Typography variant="subtitle2">
            {t('logo_text')}
          </Typography>
        </div>
        <Typography variant="h1">
          {t('title')}
        </Typography>
        <Typography variant="body1" paragraph>
          {t('paragraph')}
        </Typography>
      </div>
      <img className={classes.heroImage} src={images.collaborateHero} alt="" />
    </div>
  );
}
