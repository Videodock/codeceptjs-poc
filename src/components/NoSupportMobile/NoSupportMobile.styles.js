import { createStyles } from '@material-ui/styles';

export default ({ spacing }) => createStyles({
  root: {
    display       : 'flex',
    flexDirection : 'column',
    alignItems    : 'center',
    justifyContent: 'center',
    height        : '100vh',
    padding       : 48,
  },
  container: {
    maxWidth: 550,
  },
  heroImage: {
    width   : '70%',
    maxWidth: 500,
  },
  logoBlock: {
    display     : 'flex',
    alignItems  : 'center',
    marginBottom: spacing(0.5),
  },
  logoIcon: {
    width      : 16,
    height     : 15,
    marginRight: spacing(0.7),
  },
});
