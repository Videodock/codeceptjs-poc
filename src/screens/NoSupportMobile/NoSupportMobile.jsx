import React, { Fragment } from 'react';
import { Helmet } from 'react-helmet';
import { useTranslation } from 'react-i18next';
import NoSupportMobileComponent from '../../components/NoSupportMobile';

export default function NoSupportMobile () {
  const { t }   = useTranslation('noSupportMobile');

  return (
    <Fragment>
      <Helmet>
        <title>{t('title')}</title>
        <meta name="description" content={t('paragraph')} />
      </Helmet>
      <NoSupportMobileComponent />
    </Fragment>
  );
}
