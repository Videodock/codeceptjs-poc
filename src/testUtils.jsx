import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { ApolloProvider } from '@apollo/client';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import client from './graphql';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>
          <Router>
            {children}
          </Router>
        </I18nextProvider>
      </ThemeProvider>
    </Provider>
  </ApolloProvider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
