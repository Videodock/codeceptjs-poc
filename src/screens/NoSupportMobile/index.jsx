import NoSupportMobile from './NoSupportMobile';

export * from './NoSupportMobile';
export default NoSupportMobile;
