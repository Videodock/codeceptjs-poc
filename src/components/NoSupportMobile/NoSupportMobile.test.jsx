import React from 'react';
import { render } from '../../testUtils';
import NoSupportMobile from './NoSupportMobile';

describe('<NoSupportMobile />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<NoSupportMobile />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
