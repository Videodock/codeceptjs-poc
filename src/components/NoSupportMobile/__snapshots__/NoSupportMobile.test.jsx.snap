// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoSupportMobile /> renders and matches snapshot 1`] = `
<div
  class="makeStyles-root-1"
>
  <div
    class="makeStyles-container-2"
  >
    <div
      class="makeStyles-logoBlock-4"
    >
      <img
        alt="Trompa collaboration campaign manager"
        class="makeStyles-logoIcon-5"
        src="[object Object]"
      />
      <h6
        class="MuiTypography-root MuiTypography-subtitle2"
      >
        Trompa collaboration campaign manager
      </h6>
    </div>
    <h1
      class="MuiTypography-root MuiTypography-h1"
    >
      Sorry, we are not yet mobile..
    </h1>
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
    >
      Please use desktop or tablet for now to create modern, playable scores from known and unknown classics.
    </p>
  </div>
  <img
    alt=""
    class="makeStyles-heroImage-3"
    src="collaborate_hero.png"
  />
</div>
`;
